modules/contact/src/ContactMaster.C

```diff
@@ -417,6 +417,9 @@ contactFormulation(const std::string & the_name)
   else if ("augmented_lagrange" == name)
     formulation = CF_AUGMENTED_LAGRANGE;
 
+  else if ("tangential_penalty" == name)
```

Comment:
> This is safer because if you write constant == variable, it won't compile if you accidentally put in a single =.

Agreed.

> Correct me if I'm wrong, but if we use a MooseEnum, I think we have to do a string comparison every time we check what its value is

I think this is wrong... you can use MooseEnums in case statements like regular enums, in which case it does an integral value comparison, but I will double-check this now.